scripts/resolve_suggestions_urls.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env python3
+
+import json
+import os
+
+from jsonpath_rw import parse
+from jsonpointer import resolve_pointer, set_pointer
+
+SUGGESTIONS_URL_BASE = "https://cdn.eq.census-gcp.onsdigital.uk/data/v4.0.0"
+
+
+def json_path_to_pointer(json_path):
+    """
+    Convert a json path string into a json pointer string.
+    :param json_path: the input data to search
+    :return: json pointer equivalent to the json path
+    """
+    json_pointer = json_path.replace("[", "").replace("]", "").replace(".", "/")
+    return f"/{json_pointer}"
+
+
+def find_suggestion_urls(data):
+    json_path = parse("$..suggestions_url")
+    for match in json_path.find(data):
+        suggestions_url = match.value
+        if "suggestions_url_root" in suggestions_url:
+            yield match
+
+
+def replace_suggestions_urls(data, matches, replacement_url_root) -> int:
+    replaced_pointer_count = 0
+    for match in matches:
```

Comment:
If this was changed to `for match in find_suggestion_urls(data):`, then matches wouldn't need to be passed (and the caller wouldn't need to know anything about the matching implementation).